nvflare/fuel/f3/streaming/byte_receiver.py

```diff
@@ -140,6 +145,26 @@ def register_callback(self, channel: str, topic: str, stream_cb: Callable, *args
 
         self.registry.set(channel, topic, Callback(stream_cb, args, kwargs))
 
+    def stop_task(self, task: RxTask, error: StreamError = None, notify=True):
+        if error:
+            log.error(f"Stream error: {error}")
+            task.stream_future.set_exception(error)
+
+            if notify:
+                message = Message()
+
+                message.add_headers(
+                    {
+                        StreamHeaderKey.STREAM_ID: task.sid,
+                        StreamHeaderKey.DATA_TYPE: StreamDataType.ERROR,
+                        StreamHeaderKey.ERROR_MSG: str(error),
+                    }
+                )
+                self.cell.fire_and_forget(STREAM_CHANNEL, STREAM_ACK_TOPIC, task.origin, message)
+
+        task.eos = True
+        self.rx_task_map.pop(task.sid)
```

Comment:
It might be safer to do self.rx_task_map.pop(task.sid, None), in case the item is already gone when you try to pop it.

Reply:
Done. Also added lock to map operations.